gif-screencast.el

```diff
@@ -133,58 +157,99 @@ various programs run here."
-                                        (gif-screencast-print-status process event))))
-          (set-process-sentinel p 'gif-screencast-print-status))
-        (when (and gif-screencast-autoremove-screenshots
-                   (eq (process-exit-status p) 'exit)
-                   (= (process-exit-status p) 0))
-          (dolist (f gif-screencast--frames)
-            (delete-file (cdr f))))))))
+    (if (memq window-system '(mac ns))
+        (gif-screencast--cropping)
+      (gif-screencast--generate-gif nil nil))))
+
+(defun gif-screencast--generate-gif (process event)
+  "Generate GIF file."
+  (when process
+    (gif-screencast-print-status process event))
+
+  (let (delays
+        (index 0)
+        (frames gif-screencast--frames))
+    (while (cdr frames)
+      (push (list "(" "-clone" (number-to-string index) "-set" "delay"
+                  ;; Converters delays are expressed in centiseconds.
+                  (format "%d" (* 100 (float-time
+                                       (time-subtract (car (cadr frames)) (caar frames)))))
+                  ")" "-swap" (number-to-string index) "+delete")
+            delays)
+      (setq index (1+ index)
+            frames (cdr frames)))
+    (message "Compiling GIF with %s..." gif-screencast-convert-program)
+    (let ((output (expand-file-name
+                   (format-time-string "output-%F-%T.gif" (current-time))
+                   (or (and (file-writable-p gif-screencast-output-directory)
+                            gif-screencast-output-directory)
+                       (read-directory-name "Save output to directory: "))))
+          p)
+      (setq p (apply 'start-process
+                     gif-screencast-convert-program
+                     (get-buffer-create gif-screencast-log)
+                     gif-screencast-convert-program
+                     (append
+                      gif-screencast-convert-args
+                      (mapcar 'cdr gif-screencast--frames)
+                      ;; Delays must come after the file arguments.
+                      (apply 'nconc delays)
+                      (list output))))
+      (if gif-screencast-want-optimized
+          (set-process-sentinel p (lambda (process event)
+                                    (if (and (eq (process-status process) 'exit)
+                                             (= (process-exit-status process) 0))
+                                        (setq p (gif-screencast-optimize output))
+                                      (gif-screencast-print-status process event))))
+        (set-process-sentinel p 'gif-screencast-print-status))
+      (when (and gif-screencast-autoremove-screenshots
+                 (eq (process-exit-status p) 'exit)
+                 (= (process-exit-status p) 0))
+        (dolist (f gif-screencast--frames)
+          (delete-file (cdr f)))))))
+
+(defun gif-screencast--cropping-region ()
```

Comment:
I like to name functions as a verb, so here it would be `git-screencast--crop-region`.

Reply:
Thanks. Reflected.

Reply:
Sorry, I got too fast here.  `cropping-region` is find then.